sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -194,5 +231,10 @@ public override long Position
                 set => throw CreateException();
             }
         }
+
+        /// <summary>
+        /// Exists as a private key entry into the <see cref="HttpMessage._typeProperties"/> dictionary for stashing string keyed entries in the Type keyed dictionary.
+        /// </summary>
+        private class PropertyKey {}
```

Comment:
Would this be better as a struct? Also, maybe we establish some naming convention for these, e.g. <PropertyName>MessagePropertyKey?